config.json

```diff
@@ -428,6 +428,18 @@
       ]
     },
     {
+      "uuid": "dd74b65c-0d26-4821-9add-064e32e3a5bd",
+      "slug": "binary-search",
+      "core": false,
+      "unlocked_by": null,
+      "difficulty": 7,
```

Comment:
any reason why difficulty 7 exercise is coming between two difficulty 4 exercise?

Reply:
I was unsure about the position in config.json, because I haven't found a canonical list. So I looked into https://github.com/exercism/ecmascript/blob/master/config.json and could not figure out the logic behind order and difficulty. So I have added binary-search before robot-simulator in the rust version, because it was before robot-simulator in the ecmascript version. If you have a better position, please suggest a change or fix it, I don't have any strong opinions about it.

Reply:
I think the issue is with the assigned difficulty, not with the position in the list. I don't think an exercise which can be implemented in 30 lines of code in under 30 minutes is of difficulty 7. If you have it in the same relative position as it was in a different language track, the correct thing would be to rate this exercise as a difficulty 4, so it fits in with its peers.

Reply:
The difficulty of 7 I had also just copied from the ecmascript version, because implementing a binary-search in Rust should should have more or less the same relative difficulty as in ecmascript. The length of the solution is in my opinion no criteria, because some exercises required just a lot of easy code while others can be solved in few lines of difficult code. It's also highly subjective. 

Actually when I look at the order/difficulty of existing tasks a lot makes subjective no sense at all for me. 
Look for example at alphametics & pig-latin. Both a have a difficulty of 4 and pig-latin comes after alphametics, but alphametics is a my view 10-times harder than pig-latin.

Reply:
> unsure about the position in config.json, because I haven't found a canonical list

Because it may differ by language, it does not seem there will be a canonical list. I wonder if https://github.com/exercism/go/issues/904 will discuss this (they have not yet, therefore you don't need to visit that link)

> alphametics & pig-latin. Both a have a difficulty of 4

I can explain alphametics. Everything in "Getting Rusty" in https://github.com/exercism/rust/blob/aea470a76a9c6f0005adebb2a00d969512da3b6e/problems.md was assigned 4. This statement is not to be construed as an endorsement or rejection of the proposition "4 is the correct difficulty".

> alphametics is a my view 10-times harder than pig-latin

I agree that alphametics is harder than pig latin. Alphametics should come later.

> suggest a change

OK, then my suggestion will be changing the difficulty to 4.